### DISCUSSION TITLE
client.realm for a newer osu!lazer than current osu!lazer should probably be handled better

### DISCUSSION THREAD

**Discussion by puyoxyz** [ppy/osu#17148] [Ideas]
I just accidentally opened a version of osu!lazer older than the last version of osu!lazer I opened, and all my skins + beatmaps were gone.

Turns out, it just renamed client.realm -> client_somenumbers_corrupt.realm + made a new client.realm, and I was able to rename the old file back to client.realm to get my skins and maps back, but still, this should probably be handled better. (the average user probably wouldn't know to do this)

i.e. store which version of osu!lazer the realm was made in, then on launch check if it's newer than the current version, and if it is show a warning before marking it as corrupt and making a new one?

**Comment by peppy** [comment]
How did you run the older version? I'd sooner consider blocking startup or showing a prompt than changing this behaviour for releases. Storing different versions and restoring sounds like recipe for serious disaster.

**Reply by puyoxyz** [reply]
> Storing different versions and restoring sounds like recipe for serious disaster.

That's not what I meant, I just meant, check if realm is too new and if it is then warn the user (let them back out and open a newer osu!lazer) before marking it as corrupt and making a new one like it does now (or yeah, blocking startup is also probably a good idea)

> How did you run the older version? 

Opened the wrong one by accident (am stupid)

**Reply by SensouShin** [reply]
it also happend to me. when you open a development build that  can be older than the normal version for plenty of reasons, it saves the current db and creates a fresh.
